thundergnat/aoc2020-05-a-and-b.raku

```diff
@@ -0,0 +1,16 @@
+my @seats = slurp.words.map: { Hash.new(
+     'row'  => :2(.comb.head(7).join.trans([<F B>] => [0,1])),
+     'seat' => :2(.comb.tail(3).join.trans([<L R>] => [0,1])),
```

Comment:
Thanks for this.  I learned both about the (obvious in retrospect) connection between binary and binary search and about the `:2` syntax for other bases, which I'd somehow missed.

Is there any particular reason you went with `.comb.head(7).join` instead of the shorter `substr(^7)`, by the way?

Reply:
Sorry, I didn't notice this comment earlier.

No particular reason, just what came to mind first. Not really trying to golf, if I was, `.comb[^7]` is shorter yet. 

Then again, I could have gotten away with not splitting the row and seat out separately at all which would be much shorter yet. It would have made part B a slightly more effort though.